api/routes/auth.js

```diff
@@ -0,0 +1,107 @@
+const router = require("express").Router();
+const Users = require("../models/user");
+const bcrypt = require("bcrypt");
+const { sign, verify } = require("jsonwebtoken");
+
+const { SECRET_KEY } = process.env;
+
+//Creates an account - validate username and password, return JWT
+router.post("/signup", async (req, res, next) => {
+  const status = 201;
+
+  try {
+    const { username, password } = req.body;
+    //Check for existing user
+    let user = await Users.findOne({ username });
+    if (user) throw new Error(`User ${username} already exists`);
+    //Check for password min length
+    if (req.body.password.length < 8)
+      throw new Error("Password must be atleast 8 characters long!");
+    const saltRounds = 10;
+    const hashedPassword = await bcrypt.hash(password, saltRounds);
+    user = await Users.create({
+      username,
+      password: hashedPassword,
+      admin: req.body.admin
+    });
+    const payload = { id: user._id };
+    const options = { expiresIn: "1 day" };
+    const token = sign(payload, SECRET_KEY, options);
+    res.status(status).json({ status, token });
+  } catch (error) {
+    console.error(error);
+    const e = new Error("Enter a valid username and password");
+    e.status = 400;
+    next(e);
+  }
+});
+
+//Login user - validates user and username-password match
+router.post("/login", async (req, res, next) => {
+  const status = 201;
+  try {
+    const { username, password } = req.body;
+    let user = await Users.findOne({ username });
+    if (!user) throw new Error(`User ${username} not found`);
```

Comment:
In this case, I would throw the same error message as the one below. Otherwise, it makes it easier for someone to figure out whether or not they "guessed" at a right username.